emission/core/wrapper/trip.py

```diff
@@ -284,6 +322,72 @@ def _save_perturbed(self, perturbeds):
             for perturbed in perturbeds:
                 perturbed.save_to_db()
 
+class Fake_Trip(Trip):
+    def __init__(self, _id, user_id, trip_id, sections, start_time, end_time, trip_start_location, trip_end_location):
+        super(Fake_Trip, self).__init__(_id, user_id, trip_id, sections, start_time, end_time, trip_start_location, trip_end_location)
+        self.mode_list = self._init_mode_list(sections)
+
+    @classmethod
+    def _init_mode_list(self, sections):
+        if not sections:
+            return None
+        mode_list = []
+        mode_set = set()
+        for section in sections:
+            mode_list.append(section.mode)
+            mode_set.add(section.mode)
+        if len(mode_set) == 1:
+            return mode_set.pop()
+        return mode_list
+
+    @classmethod
+    def _init_sections(cls, user_id, trip_id, num_sections):
+        sections = []
+        db = edb.get_fake_sections_db()
+        json_object = db.find({'user_id': user_id, 'trip_id' : trip_id}, limit = num_sections)
+        for section_json in json_object:
+            sections.append(Section.section_from_json(section_json))
+        return sections
+
+
+
+    def save_to_db(self):
+        db = edb.get_fake_trips_db()
+        db.insert({"_id": self._id, "user_id": self.user_id, "trip_id": self.trip_id, "sections": range(len(self.sections)), "trip_start_datetime": self.start_time,
+                "trip_end_datetime": self.end_time, "trip_start_location": self.trip_start_location.coordinate_list(), 
+                "trip_end_location": self.trip_end_location.coordinate_list(), "mode_list": self.mode_list})
+
+    @classmethod
+    def trip_from_json(cls, jsn):
```

Comment:
Isn't this very similar to the method in Trip? Why do you have to copy and paste in here? Can't you just call the method in the superclass? Copy-pasting without a strong reason is a set of errors waiting to happen...